client/a8c-for-agencies/sections/referrals/hooks/use-fetch-referrals.ts

```diff
@@ -0,0 +1,49 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+import { useSelector } from 'calypso/state';
+import { getActiveAgencyId } from 'calypso/state/a8c-for-agencies/agency/selectors';
+import type { Referral, ReferralAPIResponse } from '../types';
+
+export const getReferralsQueryKey = ( agencyId?: number ) => {
+	return [ 'a4a-referrals', agencyId ];
+};
+
+const getClientReferrals = ( referrals: ReferralAPIResponse[] ) => {
+	const clients: Referral[] = [];
```

Comment:
Just a nitpick, we could use a hash map here, with `client_id` as a key. But I'm not sure how much elements can we have an if it would be any problem for performance.